### ISSUE TITLE
[Bug]: v7_partialHydration with lazy routes result in empty route

### ISSUE THREAD

**Issue by lasseklovstad** [remix-run/react-router#11120]
### What version of React Router are you using?

6.21.0

### Steps to Reproduce

1. Create a new router with `RouterProvider`
2. Add a lazy route
3. Add feature flag `v7_partialHydration=true`

```typescript
let router = createBrowserRouter(
  [
    {
      path: '/',
      lazy: () => import('./test'),
    },
  ],
  { future: { v7_partialHydration: true } }
);

export default function App() {
  return <RouterProvider router={router} />;
}

// test.tsx
export const Component = () => {
  return <div>Hello, I am lazy</div>;
};
```
Reproduction:
https://stackblitz.com/edit/github-bzxwek?file=src%2Fapp.tsx

### Expected Behavior

As of my understanding feature flags is the default behavior in the next version so I want to set this flag to true even though this feature (partial hydration) should only apply to server rendered apps?

Anyways, when setting this flag to true it is expected not to break lazy routes.

### Actual Behavior

lazy route is not rendered and warning in console: Matched leaf route at location "/" does not have an element or Component. This means it will render an <Outlet /> with a null value by default resulting in an "empty" page.

**Comment by brophdawg11** [comment]
> I want to set this flag to true even though this feature (partial hydration) should only apply to server rendered apps

Yep!  That's right - this is definitely a bug as `lazy` routes should work fine in SPA's with `partialHydration` specified.  Also worth noting, if you previously had a `fallbackElement`, you'd want to move that to a `HydrateFallback``/`hydrateFallbackElement` on a root route (or deeper if you wish).  `HydrateFallback` is basically a more powerful `fallbackElement` since it can be specified at any level of the tree.

**Comment by brophdawg11** [comment]
This is fixed in #11121 and will be available in the next release 👍
